File: Implementation/yunseong/4_1.js
// 4.1
// time complexity : O(n)
// input file logic
const fs = require("fs");
const input = fs.readFileSync("./test.txt").toString().trim().split("\n");
const n = +input[0];
const direction = input[1].trim().split(" ");

const dx = [-1, 1, 0, 0]; // direction x
const dy = [0, 0, 1, -1]; // direction y
// indicate based by input sign
const dir = {
  U: 0,
  D: 1,
  R: 2,
  L: 3,
};

// start point (1,1)
let point = {
  x: 1,
  y: 1,
};

// check next point is out of range
const isOutBound = (nextX, nextY) => {
  return nextX < 1 || nextX > n || nextY < 1 || nextY > n;
};

// iterate input direction
direction.forEach((arrow) => {
  const nextX = point.x + dx[dir[arrow]];
  const nextY = point.y + dy[dir[arrow]];
  if (!isOutBound(nextX, nextY)) {
    point.x = nextX;
    point.y = nextY;
  }
});

console.log(`${point.x} ${point.y}`);

File: Implementation/4-1.py
n = int(input())

###입력키 리스트 받기
key = list(map(str,input().split()))

x = 1
y = 1

for i in key:
    ###사각형 크기보다 작을 때 수행
    if(x < n and y < n):
        if(i == "R"):
            x += 1
        elif(i == "L"):
            x -= 1
        elif(i == "U"):
            y -= 1
        elif(i == "D"):
            y += 1
    ###사각형 크기보다 크면 넘어가기
    else:
        continue
    ###좌표가 0일 때 처리
    if (x == 0):
        x += 1
    elif (y == 0):
        y += 1

print(y, x)